gateway-ha/src/test/java/io/trino/gateway/ha/TestGatewayHaSingleBackend.java

```diff
@@ -18,8 +28,11 @@
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.TestInstance;
 import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.junit.jupiter.api.extension.ExtendWith;
 
 @TestInstance(Lifecycle.PER_CLASS)
+@ExtendWith(DropwizardExtensionsSupport.class)
+@Slf4j
 public class TestGatewayHaSingleBackend {
```

Comment:
question: is the goal of this test case to be a sub-system test?


my thinking is: if we just wanted to test that the gateway chooses backends properly based on health as a single unit you could replace:
```
    SECONDS.sleep(WAIT_FOR_BACKEND_IN_SECONDS);
```

with:

```
    HaGatewayLauncher haGatewayLauncher = new HaGatewayLauncher("io.trino");
    haGatewayLauncher.run(args);

    // Now populate the backend
    setUpBackend(
            "trino1", "http://localhost:" + backendPort, "externalUrl", true, "adhoc", routerPort);

    // When a backend becomes active for the first time OR is transitioned from inactive to active it will not yet be
    // treated as healthy. Without a real health check happening we will extract the RoutingManager and set them to healthy manually
    Field privateInjectorField = BaseApp.class.
            getDeclaredField("injector");
    privateInjectorField.setAccessible(true);

    Injector injector = (Injector) privateInjectorField.get(haGatewayLauncher);
    injector.getInstance(RoutingManager.class).updateBackendHealth("trino1", BackendHealth.HEALTHY);
```

This hack to grab the class with reflection is just to keep the test as close to the module as possible (and do without a sleep pause). Curious on your thoughts for it.

If the goal is to cover as much repository as possible in the test as sub-system instead of integration then fair enough 👍

Reply:
great idea! included this in my new PR